### DISCUSSION TITLE
How to implement a clickable textblock WPF MVVM

### DISCUSSION THREAD

**Discussion by HowardPWeiss** [unoplatform/uno#12502] [Q&A]
I am working on a WPF MVVM application using UnoPlatform for Linux on Ubuntu 20.04 using Visual Studio code and c#

I have created an ObservableCollection of TextBlocks which I populate from my ViewModel.  I would like to be able to click on any of these textblocks and invoke a method in my ViewModel.  I am hoping that one of the arguments to the event handler will identify which textblock object was checked

Here is my current XAML to define the textblock objects

       <DataTemplate x:Key="TextBlockObject">
            <TextBlock Width="{Binding Width}" Height="{Binding Height}" 
                Text="{Binding Text}" Foreground="{Binding Foreground}" 
                Name="Cell_0_1_0" FontSize="{Binding FontSize}"
                Visibility="{Binding Path=IsVisible, Converter={StaticResource BoolToVisibilityConverter}}"
                TextAlignment="{Binding TextAlignment}"
                >
                <TextBlock.RenderTransform>
                    <TranslateTransform X="{Binding X}" Y="{Binding Y}"/>
                </TextBlock.RenderTransform>
            </TextBlock>

and  here is the XAML to define the textblock items

    <ItemsControl ItemsSource="{Binding ObjectCollection}" ItemTemplateSelector="{StaticResource DTS}">
            <ItemsControl.ItemsPanel>
                <ItemsPanelTemplate>
                    <Canvas x:Name="Gameboard" Margin="13,105"  Width="800" Height="922" Background="Red"  VerticalAlignment="Top" HorizontalAlignment="Left">
                        <Rectangle Top="100" Left ="400" Width="150" Height="150" Stroke="Red" Fill="Gray" StrokeThickness="2">
                        </Rectangle>
                    </Canvas>
                </ItemsPanelTemplate>
            </ItemsControl.ItemsPanel>
        </ItemsControl>

I am hoping that I can an event handler for a left mouse click in the DataTemplate for the text block objects.

I have been looking on the internet and I am thoroughly confused on how to do this.  In particular, I have not found an example of how do this in when using the UnoPlatform

Any help would be appreciated

Howard Weiss

**Comment by jeromelaban** [comment]
`TextBlock` supports [`Hyperlink`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.documents.hyperlink?view=winrt-22621), which is what you seem to be looking for. You can also register a `Tap` event handler on the hyperlink.
